File: Hanoi.py
"""
This is the code for the classic tower of hanoi problem in Python.
It is famous for utilising dynamic programming.
这是经典汉诺塔问题的 Python 代码。
它因运用递归算法而闻名。
"""

def Hanoi(n, A, B, C):
    if n == 1:
        print(f"Move plate {n} from disk {A} to disk {C}")
    else:
        Hanoi(n-1, A, C, B)
        print(f"Move plate {n} from disk {A} to disk {C}")
        Hanoi(n-1, B, A, C)
        
n = int(input("Please input the number of disks you want to move: "))
Hanoi(n, 'A', 'B', 'C')

File: Hanoi.java
/* This is the code for the classic tower of hanoi problem in Java.
It is famous for utilising dynamic programming.
*/

import java.util.Scanner;

public class Hanoi {
    public static void Hanoi(int n, char A, char B, char C){
        if (n == 1) {
            System.out.printf("Move plate %d from %c to %c\n", n, A, C);
        } else {
            Hanoi(n-1, A, C, B);
            System.out.printf("Move plate %d from %c to %c\n", n, A, C);
            Hanoi(n-1, B, A, C);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Please input the number of disks you want to move: ");
        int n = scanner.nextInt();
        Hanoi(n, 'A', 'B', 'C');
    }
}